node/src/types/deploy.rs

```diff
@@ -558,11 +558,17 @@ impl Deploy {
         payment: ExecutableDeployItem,
         session: ExecutableDeployItem,
         secret_key: &SecretKey,
+        account: Option<PublicKey>,
```

Comment:
Personally, I think `account` should be mandatory, `new()` should not even take the `secret_key` and deploys should be signed separately with `sign` method. Those should be the building blocks. From there, we could build everything else.

Reply:
As discussed offline, we'll stick with the planned approach as implemented in this PR for now, and iterate on that as and if required.